lib_nn/api/AggregateFn.hpp

```diff
@@ -0,0 +1,578 @@
+#ifndef LIB_NN_AGGREGATE_FN_HPP_
+#define LIB_NN_AGGREGATE_FN_HPP_
+
+#include <array>
+#include <vector>
+
+#include "Utils.hpp"
+#include "geom/Filter2dGeometry.hpp"
+#include "vpu.hpp"
+#include "xs3_vpu.h"
+
+namespace nn {
+
+class AggregateFn {
+ public:
+  /**
+   * @brief Function to aggregate an input receptive field as defined by the
+   * operation it is performing, i.e. convolution, maximum, addition, etc.
+   * Arrgreation is preformed in batches of channel groups, each of the same
```

Comment:
Check syntax in oxygen comment.

Reply:
what do you mean?

Reply:
* Aggregation is preformed in batches of